### PR TITLE
crackql #3486

### PR DESCRIPTION
**Pull request by dualfade**
please review --

### CODE REVIEW

File: packages/crackql/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+depends=('python' 'python-virtualenv')
+makedepends=('git' 'python-pip')
+source=("git+https://github.com/nicholasaleks/$_pkgname.git")
+sha512sums=('SKIP')
+install="$pkgname.install"
+
+pkgver() {
+  cd $_pkgname
+
+  echo $(git rev-list --count HEAD).$(git rev-parse --short HEAD)
```
Comment (noraj) [inline packages/crackql/PKGBUILD:22]: Use this https://github.com/BlackArch/blackarch-pkgbuilds/blob/4ae2d3f269ad76eda2d78f3ea172e9bba6fd1835/PKGBUILD-generic-git#L24 when there are tags available

File: packages/crackql/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+pkgname=crackql
+_pkgname=CrackQL
+pkgver=53.f4d41cd
+pkgrel=1
+pkgdesc='GraphQL password brute-force and fuzzing utility'
+arch=('any')
+groups=('blackarch' 'blackarch-webapp' 'blackarch-exploitation' 'blackarch-fuzzer')
+url='https://github.com/nicholasaleks/CrackQL'
+license=('BSD 3-Clause License')
```
Comment (noraj) [inline packages/crackql/PKGBUILD:12]: I think you should just tag BSD as per https://wiki.archlinux.org/title/PKGBUILD#license

File: packages/crackql/crackql.install [hunk 3]
```diff
@@ -0,0 +1,15 @@
+post_install() {
+  set -e
+  cd /usr/share/crackql
+  python -m virtualenv venv
+  source venv/bin/activate &&
+    pip install --isolated --root="/usr/share/crackql" --prefix='venv' \
+      'certifi' 'charset-normalizer' 'graphql-core' 'idna' 'jinja2' \
```
Comment (noraj) [inline packages/crackql/crackql.install:7]: Why don't you pull them from https://github.com/nicholasaleks/CrackQL/blob/master/requirements.txt as per https://github.com/BlackArch/blackarch-pkgbuilds/blob/master/python-standalone.install?

File: packages/crackql/crackql.install [hunk 4]
```diff
@@ -0,0 +1,15 @@
+post_install() {
+  set -e
+  cd /usr/share/crackql
+  python -m virtualenv venv
```
Comment (noraj) [inline packages/crackql/crackql.install:4]: Please just use the native venv as per https://github.com/BlackArch/blackarch-pkgbuilds/blob/4ae2d3f269ad76eda2d78f3ea172e9bba6fd1835/python-standalone.install#L4

File: packages/crackql/PKGBUILD [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+depends=('python' 'python-virtualenv')
```
Comment (noraj) [inline packages/crackql/PKGBUILD:13]: Please remove python-virtualenv and use venv instead in .install.

**Review by noraj — CHANGES_REQUESTED** [review]

File: packages/crackql/crackql.install (hunk 3 above)
Comment (dualfade) [reply packages/crackql/crackql.install:7]: Packages and run fine, however I do know get this error:
ERROR: file:///usr/share/crackql does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

~!>D

Comment (dualfade) [reply packages/crackql/crackql.install:7]: I can make a request to the devs to create proper project.toml file. Let me know.

Comment (dualfade) [reply packages/crackql/crackql.install:7]: fyi. I talked to the devs. They will add a setup file. 
Also his new book coming out --
https://nostarch.com/black-hat-graphql

That's awesome.

Comment (noraj) [reply packages/crackql/crackql.install:7]: https://github.com/nicholasaleks/CrackQL/issues/8

**Pull request closed without merging**